src/module/elements.rs

```diff
@@ -0,0 +1,185 @@
+//! Table elements within a wasm module.
+
+use crate::error::Result;
+use crate::module::emit::{Emit, IdsToIndices};
+use crate::module::functions::FunctionId;
+use crate::module::tables::TableId;
+use crate::module::Module;
+use crate::passes::Used;
+use id_arena::{Arena, Id};
+use parity_wasm::elements;
+use std::collections::HashMap;
+
+/// A passive element segment identifier
+pub type ElementId = Id<Element>;
+
+/// A passive segment which contains a list of functions
+#[derive(Debug)]
+pub struct Element {
+    members: Vec<FunctionId>,
```

Comment:
Not a set because order is important, right?

Maybe we *also* want to maintain a set for if someone wants to add a new function, and can instead just reuse and existing member? Eh, probably not worth it till we need it...

Reply:
Correct yeah, this is like a data segment which is a blob of bytes and can be copied to/from.

I'll rename this to have `Passive` in the name as I think that makes more sense

Reply:
Hm actually not a huge fan of `Passive`, so left as-is.